vexim/siteaddsubmit.php

```diff
@@ -62,10 +62,21 @@
     die;
   }
   if(isset($_POST['maildir']) && isset($_POST['localpart'])) {
-    $smtphomepath = realpath($_POST['maildir'] . "/") .
-      "/" . $_POST['domain'] . "/" . $_POST['localpart'] . "/Maildir";
-    $pophomepath = realpath($_POST['maildir'] . "/") .
-      "/" . $_POST['domain'] . "/" . $_POST['localpart'];
+    if (substr($_POST['maildir'], 0, 1) !== '/') {
+      header ("Location: site.php?failmaildirnonabsolute={$_POST['maildir']}");
+	  die();
+    }
+    if ($testmailroot && is_dir(realpath($_POST['maildir'])) === false) {
```

Comment:
I think you misunderstood the code, or my intentions.

First of all,
`if ($testmailroot && is_dir(realpath($_POST['maildir'])) === false)`
is equivalent to
`if ($testmailroot && (is_dir(realpath($_POST['maildir'])) === false))`
and is more or less the same as your last suggestion.

Second, if I remember correctly, `is_dir()` will not even be run if `$testmailroot` is `false`, because in that case the result of `&&` is already known to be `false`. So yeah, `realpath()` shouldn't be executed if the setting is false.

Now, your second example is actually incorrect: I do not want to trigger `header()` and `die()` if `$testmailroot` is `false`, regardless of whether `realpath()` fails or succeeds. That's the whole point of that setting! :)